### PR TITLE
feat(updater): Enhance README with description of update process 

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
### Proposed Change
* Enhance the Updater readme with a high-level description of the update process

I tried to just hit the overall bullet points of how this process works. I tried not to get too tangled up in all the failure scenarios, all though I do try to mention some when I think they are pertinent.

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: updater/README.md [hunk 1]
```diff
@@ -2,4 +2,50 @@
...
+## Updating Overview
+
+1. The collector receives PackagesAvailable message.
+2. The collector downloads tarball containing new updater and updated artifacts (e.g. new collector, plugins, etc.) based on the contents of the PackagesAvailable message.
+3. The collector unpacks tarball into `$INSTALL_DIR/tmp/latest`.
+4. The collector copies the newest updater binary from `$INSTALL_DIR/tmp/latest` to the working directory.
+5. The collector starts the updater in as a separate process in a new process group.
+  a. If the updater fails to stop the collector within 30 minutes, 
+
+6. The updater starts, then shuts down collector through the service manager.
```
Comment (schmikei) [inline updater/README.md:16]:
```suggestion
6. The updater starts, then shuts down the collector through the service manager.
```
think this makes it read a tad better!

**Review by schmikei — APPROVED** [review]
Overall looks good, would like someone who knows the updater a tad better to fact check any gaps I have.

**Review by JonathanWamsley — COMMENTED** [review]
The format with the a. seems kinda weird to me. Is this intended? 
pic

**Comment by BinaryFissionGames** [comment]
> The format with the a. seems kinda weird to me. Is this intended?

Definitely not intentional, good catch! I'll make these bullet points, hopefully that looks better.

**Review by cpheps — APPROVED** [review]

File: updater/README.md [hunk 2]
```diff
@@ -2,4 +2,50 @@
...
+
+1. The collector receives PackagesAvailable message.
+2. The collector downloads tarball containing new updater and updated artifacts (e.g. new collector, plugins, etc.) based on the contents of the PackagesAvailable message.
+3. The collector unpacks tarball into `$INSTALL_DIR/tmp/latest`.
+4. The collector copies the newest updater binary from `$INSTALL_DIR/tmp/latest` to the working directory.
+5. The collector starts the updater in as a separate process in a new process group.
+   * If the updater fails to stop the collector within 30 seconds, the collector will kill the updater and abort the update. 
+
+6. The updater starts, then shuts down the collector through the service manager.
+7. The collector shuts down, orphaning the updater process.
```
Comment (StefanKurek) [inline updater/README.md:17]: minor, but step 7 sounds partly like it's repeating part 6. maybe "The updater process is orphaned when the collector shuts down". either that or combine them both into one step.

**Review by StefanKurek — APPROVED** [review]

**Pull request merged**